src/features/core/deviceModeTransformation/transformationHandler.js

```diff
@@ -61,67 +62,69 @@ class TransformationsHandler {
               const { status } = xhr;
               let { response } = xhr;
 
-              if (status === 200) {
-                if (response && typeof response === 'string') {
-                  response = JSON.parse(response);
-                } else {
-                  reject(`[Transformation]:: Transformation failed. Invalid response from server.`);
+              switch (status) {
+                case 200: {
+                  if (response && typeof response === 'string') {
```

Comment:
What if the response is undefined or not a string?

Reply:
That is the contract we have with the server. 200 will have some response in correct format

Reply:
In that case, there is no need for the 'if' check.

Reply:
Adressed